### ISSUE TITLE
vector installed via homebrew should look for configuration at `/usr/local/etc/vector/vector.toml`

### ISSUE THREAD

**Issue by jszwedko** [timberio/vector#7719]
Noted by user in discord: https://discord.com/channels/742820443487993987/746070604192415834/848122672029761536

When installing `vector` via `brew`, the default configuration file is shipped as `/usr/local/etc/vector/vector.toml` since homebrew puts everything under `/usr/local`. However, the `vector` binary still looks for it at `/etc/vector/vector.toml`.

When `vector` is installed via homebrew, it should look for the configuration to be in `/usr/local/etc/vector/vector.toml` by default.

Also, it should probably default the data dir to `/usr/local/var/lib/vector`.

This is slightly related to #7108

**Comment by jszwedko** [comment]
A user reported this again today however the path was different for them. For them, it installs the default config file at:

```
~/.vector/config/vector.toml
```

I think we could update Vector to look in several places by default.

**Comment by JeanMertz** [comment]
I've also thought about this before. I prefer configuration files to look by default (or as part of a number of default places) in `XDG_CONFIG_HOME`, which defaults to `~/.config`, so for Vector this would be `${XDG_CONFIG_HOME}/vector/vector.toml`.

Ideally, what would happen is:

1. Install a default config into `/usr/local/etc/vector/vector.toml`
2. If I run Vector without any arguments, it should use that file
3. If I manually create a file in `$XDG_CONFIG_HOME/vector`, and then run vector without any arguments, it should use the file in `XDG_CONFIG_HOME` instead.

In other words, it should search a list of folders to find a Vector config file, and I think those should be (in order):

1. `$XDG_CONFIG_HOME/vector`
2. `$HOME/.vector`
3. `/usr/local/etc/vector` OR `/etc/vector`
